mcs/class/monodoc/Resources/mdoc-html-utils.xsl

```diff
@@ -2336,27 +2336,20 @@
 				<xsl:if test="contains($membername, '&lt;')">
 					<xsl:value-of select="substring-before ($membername, '&lt;')" />
 				</xsl:if>
-				<xsl:text>``</xsl:text>
-				<xsl:value-of select="$numgenargs" />
+				<xsl:value-of select="'{'"/>
+					<xsl:for-each select="TypeParameters/TypeParameter">
+						<xsl:if test="not(position()=1)">, </xsl:if>
+						<xsl:value-of select="@Name"/>
+					</xsl:for-each>
+				<xsl:value-of select="'}'"/>
 			</xsl:otherwise>
 		</xsl:choose>
 	</xsl:template>
-
+    
 	<xsl:template name="GetEscapedTypeName">
 		<xsl:param name="typename" />
-		<xsl:variable name="base" select="substring-before ($typename, '&lt;')" />
-
-		<xsl:choose>
-			<xsl:when test="$base != ''">
-				<xsl:value-of select="translate ($base, '+', '.')" />
-				<xsl:text>`</xsl:text>
-				<xsl:call-template name="GetGenericArgumentCount">
-					<xsl:with-param name="arglist" select="substring-after ($typename, '&lt;')" />
-					<xsl:with-param name="count">1</xsl:with-param>
-				</xsl:call-template>
-			</xsl:when>
-			<xsl:otherwise><xsl:value-of select="translate ($typename, '+', '.')" /></xsl:otherwise>
-		</xsl:choose>
+		<!-- no longer escaping as it was causing invalid links to be generated -->
```

Comment:
Comments are generally to be avoided; that's what commit messages are for, which can also provide more context. This comment is also meaningless; "going forward", this is a 3-line template, one of which is a comment that describes a past which isn't available w/o `git log` *anyway*...